lib/pbench/agent/tool_meister.py

```diff
@@ -248,162 +348,133 @@ def wait(self):
-                " 30 seconds",
-                self.name,
-            )
-        else:
-            if sts != 0 and sts != -(signal.SIGTERM):
-                self.logger.warning(
-                    "Transient tool %s start process failed with %d", self.name, sts
-                )
-        self.start_process = None
+        # First wait for the "stop" process to do it's job ...
+        self.stop_process = self._wait_for_process_w_kill(self.stop_process, "stop")
+        # ... then we wait for the start process to finish; in either case,
+        # we'll only wait a short time before killing them.
+        self.start_process = self._wait_for_process_w_kill(self.start_process, "start")
 
 
-class PcpTransTool(Tool):
-    """PcpTransTool - A more traditional alternative to the pcp persistent tool that
-                      allows one to run both the pmlogger and pmcd locally on each
-                      registered node. Additionally, this tool starts, stops, and
-                      sends data alongside other transient tools, rather than always
-                      running in the background.
+class PcpTransientTool(Tool):
+    """The transient tool alternative to the PCP persistent tool, which starts
+    and stops both a pmcd and pmlogger process, and sends data remotely, as
+    directed for transient tools.
     """
 
-    def __init__(self, name, tool_opts, logger=None, **kwargs):
-        super().__init__(name, tool_opts, logger=logger, **kwargs)
-        if self.tool_dir:
-            self.tool_dir = self.tool_dir / self.name.replace("-transient", "")
+    def __init__(self, name, tool_opts, **kwargs):
+        super().__init__(name, tool_opts, **kwargs)
+        self.tool_inst_dir = self.tool_dir / self.name.replace("-transient", "")
         if "/usr/libexec/pcp/bin" not in os.environ["PATH"]:
-            os.environ["PATH"] += os.pathsep + "/usr/libexec/pcp/bin"
-        pmcd_path = find_executable("pmcd")
-        if pmcd_path:
-            self.pmcd_args = [
-                pmcd_path,
-                "--foreground",
-                "--socket=./pmcd.socket",
-                "--port=55677",
-                f"--config={self.pbench_install_dir}/templates/pmcd.conf",
-            ]
-        else:
-            self.pmcd_args = None
-        pmlogger_path = find_executable("pmlogger")
-        if pmlogger_path:
-            self.pmlogger_args = [
-                pmlogger_path,
-                "--log=-",
-                "--report",
-                "-t",
-                "3s",
-                "-c",
-                f"{self.pbench_install_dir}/templates/pmlogger.conf",
-                "--host=localhost:55677",
-                f"{self.tool_dir}/%Y%m%d.%H.%M",
-            ]
-        else:
-            self.pmlogger_args = None
+            # FIXME - Shouldn't this be provided by the environment?
+            os.environ["PATH"] += ":/usr/libexec/pcp/bin"
```

Comment:
> Shouldn't this be provided by the environment?

This is just providing a fall-back -- that is, this won't be used unless `find_executable()` couldn't find it in any of the directories in the existing `PATH`.

Ideally, we would use `find_executable()` on the existing `PATH` and then, only if that came up empty, we would see if there was a suitable file at this fallback location, and so forth.  The code here does basically the same thing only in a more straightforward way.  (To be honest, since it won't hurt if the directory is in the `PATH` more than once, I wouldn't bother with the `if` statement -- just append it and be done with it.)

Reply:
Yes, this is a bit ugly.  Let me take a stab at addressing this in another commit.  At least this fixes it to work as expected, `:` vs `os.pathsep`.

Reply:
Let's take care of this in PR #2697.

Reply:
What was wrong with using `os.pathsep`?  (Not that we need it, but that seems like it would be more portable....)

Reply:
My bad, mistaked it for `os.path.sep`.